UndertaleModTool/MainWindow.xaml.cs

```diff
@@ -3093,135 +3095,156 @@ public async void UpdateApp(SettingsWindow window)
                     }
-                    File.WriteAllText(Path.Combine(updaterFolderTemp, "actualAppFolder"), ExePath);
-
-                    window.UpdateButtonEnabled = true;
-
-                    this.ShowMessage("UndertaleModTool will now close to finish the update.");
 
-                    Process.Start(new ProcessStartInfo(Path.Combine(updaterFolderTemp, "UndertaleModToolUpdater.exe"))
+                    // Move back to UI thread to perform final actions
+                    Dispatcher.Invoke(() =>
                     {
-                        WorkingDirectory = updaterFolderTemp
-                    });
+                        this.ShowMessage("UndertaleModTool will now close to finish the update.");
 
-                    CloseOtherWindows();
+                        // Invoke updater
+                        Process.Start(new ProcessStartInfo(Path.Combine(updaterFolderTemp, "UndertaleModToolUpdater.exe"))
+                        {
+                            WorkingDirectory = updaterFolderTemp
+                        });
 
-                    Closing -= DataWindow_Closing; // disable "on window closed" event handler
-                    Close();
+                        CloseOtherWindows();
+
+                        Closing -= DataWindow_Closing; // disable "on window closed" event handler
+                        Close();
+                    });
                 });
+            }
+            catch (Exception e)
+            {
+                string errMsg;
 
-                _ = Task.Run(() =>
+                if (e.InnerException?.InnerException is Exception ex)
                 {
-                    while (!end)
+                    if (ex.Message.StartsWith("Unable to read data")
+                        && e.InnerException.Message.StartsWith("The SSL connection could not be established"))
                     {
-                        try
+                        errMsg = "Failed to download new version of UndertaleModTool.\n" +
+                                 "Error - The SSL connection could not be established.";
+
+                        bool isWin7 = Environment.OSVersion.Version.Major == 6;
+                        string win7upd = "\nProbably, you need to install Windows update KB2992611.\n" +
+                                         "Open the update download page?";
+
+                        if (isWin7)
                         {
-                            UpdateProgressStatus($"Downloaded MB: {downloaded}");
-                        }
-                        catch {}
+                            if (this.ShowQuestion(errMsg + win7upd, MessageBoxImage.Error) == MessageBoxResult.Yes)
+                                OpenBrowser("https://www.microsoft.com/en-us/download/details.aspx?id=44622");
 
-                        Thread.Sleep(100);
+                            window.UpdateButtonEnabled = true;
+                            return;
+                        }
                     }
```

Comment:
can we just yoink this whole case?
We dont support win7, and if you have an SSL error then youll see that anyway.

Reply:
Yoink!